source/Digitisers/src/DDPlanarDigiProcessor.cc

```diff
@@ -308,10 +348,46 @@ void DDPlanarDigiProcessor::processEvent( LCEvent * evt ) {
           continue; 
         }
       }
+
+      //***************************************************************
+      // Smear time of the hit and apply the time window cut if needed
+      //***************************************************************
       
-      //**************************************************************************
-      // Try to smear the hit but ensure the hit is inside the sensitive region
-      //**************************************************************************
+      double hitT = simTHit->getTime();
+      
+      // Smearing time of the hit
+      if (_resT.size() and _resT[0] > 0.0) {
+      float resT = _resT.size() > 1 ? _resT.at(layer) : _resT.at(0); 
+      double tSmear  = resT > 0.0 ? gsl_ran_gaussian( _rng, resT ) : 0.0;
+      _h[hT]->Fill( resT > 0.0 ? tSmear / resT : 0.0 );
+      _h[diffT]->Fill( tSmear );
+
+      hitT += tSmear;
+      streamlog_out(DEBUG3) << "smeared hit at T: " << simTHit->getTime() << " ns to T: " << hitT << " ns according to resolution: " << resT << " ns" << std::endl;
+      }
+     
+      // Correcting for the propagation time
+      if (_correctTimesForPropagation) {
+        double dt = oldPos.r() / ( TMath::C() / 1e6 );
+        hitT -= dt;
+        streamlog_out(DEBUG3) << "corrected hit at R: " << oldPos.r() << " mm by propagation time: " << dt << " ns to T: " << hitT << " ns" << std::endl;
+      }
+      
+      // Skipping the hit if its time is outside the acceptance time window
+      if (_useTimeWindow) {
+        float timeWindow_min = _timeWindow_min.size() > 1 ? _timeWindow_min.at(layer) : _timeWindow_min.at(0);
+        float timeWindow_max = _timeWindow_max.size() > 1 ? _timeWindow_max.at(layer) : _timeWindow_max.at(0);
+        if ( hitT < timeWindow_min || hitT > timeWindow_max ) {
+        streamlog_out(DEBUG4) << "hit at T: " << simTHit->getTime() << " smeared to: " << hitT << " is outside the time window: hit dropped"  << std::endl;
+        ++nDismissedHits;
+        }
+        continue; 
```

Comment:
The `continue` has to be inside the if before. (and the statements in the if need to be indented)